src/scgpt/cell_type_annotation/script.py

```diff
@@ -0,0 +1,216 @@
+
+logger.info("Reading in data")
+# Read in data
+mdata = mu.read(par["input"])
+input_adata = mdata.mod[par["modality"]]
+adata = input_adata.copy()
+
+# Fetch batch ids for domain-specific batch normalization
+if par["dsbn"] and not par["obs_batch_label"]:
+    raise ValueError("When dsbn is set to True, you are required to provide batch labels (obs_batch_labels).")
+elif par["dsbn"] and par["obs_batch_label"]:
+    logger.info("Fetching batch id's for domain-specific batch normalization")
+    batch_id_cats = adata.obs[par["obs_batch_label"]].astype("category")
+    batch_id_labels = batch_id_cats.cat.codes.values
+    batch_ids = batch_id_labels.tolist()
+    batch_ids = np.array(batch_ids)
+    num_batch_types = len(set(batch_ids))
+elif not par["dsbn"]:
+    # forward pass requires a tensor as input
+    batch_ids = np.array([0] * adata.shape[0])
+
+logger.info("Loading model vocabulary")
+# Vocabulary configuration
+special_tokens = [par["pad_token"], "<cls>", "<eoc>"]
+logger.info(f"Loading model vocab from {par['model_vocab']}")
+vocab_file = par["model_vocab"]
+vocab = GeneVocab.from_file(vocab_file)
+[vocab.append_token(s) for s in special_tokens if s not in vocab]
+vocab.set_default_index(vocab[par["pad_token"]])
+ntokens = len(vocab)
+
+logger.info("Loading model and configurations")
+# Load model configurations
+model_config_file = par["model_config"]
+with open(model_config_file, "r") as f:
+    model_configs = json.load(f)
+embsize = model_configs["embsize"]
+nhead = model_configs["nheads"]
+d_hid = model_configs["d_hid"]
+nlayers = model_configs["nlayers"]
+
+# Instantiate model
+model = TransformerModel(
+    ntokens,
+    d_model=embsize,  # self.encoder (GenEncoder), self.value_encoder (ContinuousValueEncoder), self.transformerencoder(TransformerEncoderLayer)
+    nhead=nhead,  # self.transformer_encoder(TransformerEncoderLayer)
+    d_hid=d_hid,  # self.transformer_encoder(TransformerEncoderLayer)
+    nlayers=nlayers,  # self.transformer_encoder(TransformerEncoderLayer), self.cls_decoder
+    nlayers_cls=3,  # self.cls_decoder
+    n_cls=par["n_cls"],  # self.cls_decoder
+    vocab=vocab,
+    dropout=par["dropout"],  # self.transformer_encoder
+    pad_token=par["pad_token"],
+    pad_value=par["pad_value"],
+    do_mvc=False,
+    do_dab=False,
+    use_batch_labels=par["dsbn"],
+    num_batch_labels=num_batch_types if par["dsbn"] else None,
+    domain_spec_batchnorm=par["dsbn"],
+    input_emb_style="continuous",
+    n_input_bins=par["n_input_bins"],
+    cell_emb_style="cls",  # hard-coded
+    use_fast_transformer=False,   #TODO: parametrize when GPU is available
+    fast_transformer_backend="flash",  #TODO: parametrize when GPU is available
+    pre_norm=False,  #TODO: parametrize when GPU is available
+)
+
+model_file = par["model"]
+try:
+    logger.info(f"Loading all model params from {model_file}")
+    model.load_state_dict(torch.load(model_file, map_location=device))
+except Exception:
```

Comment:
Could you only catch the specific error here? This will catch almost everthing, even ImportError and MemoryError (which I think for sure we do not want to catch).  This is a usefull reference: https://docs.python.org/3/library/exceptions.html#exception-hierarchy

You can also capture several exception classes with something like `except (TypeError, ValueError)